src/Text/Pandoc/Readers/Markdown.hs

```diff
@@ -1732,6 +1733,42 @@ source = do
 linkTitle :: PandocMonad m => MarkdownParser m Text
 linkTitle = quotedTitle '"' <|> quotedTitle '\''
 
+-- Github wiki style link, with optional title
+-- syntax documented under https://help.github.com/en/github/building-a-strong-community/editing-wiki-content
+githubWikiLink :: PandocMonad m => MarkdownParser m (F Inlines)
+githubWikiLink = try $ do
+  guardEnabled Ext_wikilinks
+  string "[["
+  inlinedContents <- option (return $ B.fromList [])
+                      $ lookAhead $ try $ mconcat <$> manyTill inline (try $ string "]]")
+  rawContents <- manyTillChar anyChar (try $ string "]]")
+  let allowedInline x = case x of
+                          B.Str _   -> True
+                          B.Space   -> True
+                          B.Link {} -> True
```

Comment:
Are all markdown links allowed?  Or just bare URIs?
If just bare URIs, then maybe a better approach would be to use a parser that just targets what we need.

Reply:
GitHub's documentation says: `[[Link Text|nameofwikipage]]`

So there are two questions.  1) What can go in "Link Text"?  Your experiments appear to reveal that only plain text is allowed, no special formatting.  2) What can go in "nameofwikipage"?  Your experiments show  that a fully qualified URL is allowed, but clearly that's not all -- names of wiki pages are also allowed.  I imagine that just about any plain text could be the name of a wiki page, so presumably we should allow e.g. `[[Link Text|This is my page]]`.  For this we should emit a link with URL `This is my page` but also a special attribute indicating that it's a wikilink, so that users can do transformations on the path.

Reply:
(OK, looks like that's what you're doing, from the tests!)

Reply:
This is not pertinent to GitHub-flavored Markdown, but it may be informative to note that wikilinks have been growing in popularity due to the explosion of note-taking and (personal) knowledge management tools, which are often based on Markdown or Markdown-ish syntaxes. Some (nonstandard) implementations allow link text to contain inline formatting, and some even allow page titles (and thus link targets) to contain both formatting and nested links.

Reply:
Anyway, seems to me we should try to simplify the parser.  Instead of parsing inlines and raw, and then trying to piece things together, why not something like

``` haskell
wikilink = try $ do
  string "[["
  firstPart <- wikiText
  (char '|' *> complexWikilink firstPart) <|> (string "]]" *> return (wikilink with firstPart as link text and (stringify firstPart) as url)

complexWikilink firstPart = do
  url <- stringify <$> wikiText
  string "]]"
  return $ wikilink with firstPart as link text and url as URL

wikiText = many ( whitespace <|> bareURL <|> str <|> endline <|> escapedChar )
```

Reply:
I have tried you suggestion, but it seems that I miss the point

Reply:
It's okay, let me merge your commit, and then I'll see if there's a way to simplify it.